tests/acceptance/bootstrap/SpacesContext.php

```diff
@@ -254,22 +254,67 @@ public function getSpaceIdByName(string $user, string $spaceName): string {
 		return $space["id"];
 	}
 
+	/**
+	 * Retrieves the remoteItem ID for a given share in the Shares space.
+	 *
+	 * @param string $user
+	 * @param string $share //share
+	 *
+	 * @return string
+	 *
+	 * @throws Exception|GuzzleException
+	 */
+	public function getSharesRemoteItemId(string $user, string $share): string {
+		$credentials = $this->featureContext->graphContext->getAdminOrUserCredentials($user);
+		$response = GraphHelper::getSharesSharedWithMe(
+			$this->featureContext->getBaseUrl(),
+			$this->featureContext->getStepLineRef(),
+			$credentials['username'],
+			$credentials['password']
+		);
+
+		$jsonBody = $this->featureContext->getJsonDecodedResponseBodyContent($response);
+
+		// Search for the folder name in the response to find the remoteItem ID
+		foreach ($jsonBody->value as $item) {
+			if (isset($item->name) && $item->name === $share) {
+				if (isset($item->remoteItem->id)) {
+					return $item->remoteItem->id;
+				} else {
+					throw new Exception("Failed to find remoteItem ID for share: $share.");
+				}//throw-- remote id id not found
+			}
+		}
+
+		//no share
+		throw new Exception("Failed to find remoteItem ID for folder: $share in Shares space.");
+	}
+
 	/**
 	 * The method finds file by fileName and spaceName and returns data of file which contains in responseHeader
 	 * fileName contains the path, if the file is in the folder
 	 *
 	 * @param string $user
 	 * @param string $spaceName
 	 * @param string $fileName
+	 * @param string|null $share
 	 *
 	 * @return ResponseInterface
 	 * @throws GuzzleException
 	 */
-	public function getFileData(string $user, string $spaceName, string $fileName): ResponseInterface {
-		$space = $this->getSpaceByName($user, $spaceName);
+	public function getFileData(string $user, string $spaceName, string $fileName, ?string $share= null): ResponseInterface {
```

Comment:
```suggestion
	public function getFileData(string $user, string $spaceName, string $fileName, ?bool $federatedShare = false): ResponseInterface {
```

Reply:
@saw-jan
Why are we setting `$federatedShare` as as boolean?